pallets/xyk-liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,927 @@
+		///
+		/// Parameters:
+		/// - `origin`: global farm's owner.
+		/// - `global_farm_id`: farm id from which yield farm should be destroyed.
+		/// - `yield_farm_id`: id of yield farm to be destroyed.
+		/// - `asset_pair`: asset pair identifying yield farm in the global farm.
+		///
+		/// Emits `YieldFarmDestroyed` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::destroy_yield_farm())]
+		pub fn destroy_yield_farm(
+			origin: OriginFor<T>,
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			asset_pair: AssetPair,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			//NOTE: don't check amm pool existance, owner must be able to stop yield farm.
+			let amm_pool_id = T::AMM::get_pair_id(asset_pair);
+
+			T::LiquidityMiningHandler::destroy_yield_farm(who.clone(), global_farm_id, yield_farm_id, amm_pool_id)?;
+
+			Self::deposit_event(Event::YieldFarmDestroyed {
+				global_farm_id,
+				yield_farm_id,
+				who,
+				asset_pair,
+			});
+
+			Ok(().into())
+		}
+
+		/// Deposit LP shares to a liq. mining.
+		///
+		/// This function transfers LP shares from `origin` to pallet's account and mint nft for
+		/// `origin` account. Minted nft represents deposit in the liq. mining.
+		///
+		/// Parameters:
+		/// - `origin`: account depositing LP shares. This account has to have at least
+		/// `shares_amount` of LP shares.
+		/// - `global_farm_id`: id of global farm to which user wants to deposit LP shares.
+		/// - `yield_farm_id`: id of yield farm to deposit to.
+		/// - `asset_pair`: asset pair identifying LP shares user wants to deposit.
+		/// - `shares_amount`: amount of LP shares user wants to deposit.
+		///
+		/// Emits `SharesDeposited` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::deposit_shares())]
+		pub fn deposit_shares(
+			origin: OriginFor<T>,
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			asset_pair: AssetPair,
+			shares_amount: Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let amm_pool_id = Self::ensure_amm(asset_pair)?;
+
+			let amm_share_token = T::AMM::get_share_token(asset_pair);
+
+			ensure!(
+				T::MultiCurrency::ensure_can_withdraw(amm_share_token, &who, shares_amount).is_ok(),
+				Error::<T>::InsufficientAmmSharesBalance
+			);
+
+			let deposit_id = T::LiquidityMiningHandler::deposit_lp_shares(
+				global_farm_id,
+				yield_farm_id,
+				amm_pool_id,
+				shares_amount,
+				Self::get_asset_balance_in_amm_pool,
+			)?;
+
+			Self::lock_lp_tokens(amm_share_token, &who, shares_amount)?;
+			T::NFTHandler::mint_into(&T::NftCollectionId::get(), &deposit_id, &who)?;
+
+			Self::deposit_event(Event::SharesDeposited {
+				global_farm_id,
+				yield_farm_id,
+				who,
+				amount: shares_amount,
+				lp_token: amm_share_token,
+				deposit_id,
+			});
+
+			Ok(())
+		}
+
+		/// This function create yield farm entry for existing deposit. LP shares are not transferred
```

Comment:
to be consistent with other comments - it should be "Create yield ..." without the "This function"